### ISSUE TITLE
An error occurred when processing GeonamesLanguageImporter : null

### ISSUE THREAD

**Issue by RaphaelFourdrilis** [gisgraphy/gisgraphy#27]
Hello,

I have been trying to import the french dataset for 2 days now, and I can't seem to figure out what goes wrong.
Each and every time I try, everything seems to go fine until the first importer task, right after everything has been retrieved:

`An error occurred when processing GeonamesLanguageImporter : null`

I didn't change anything important in the setting files, just added the correct pgsql user / pass, went through the basic installation process, ran the importer by selecting only France and all POI types, left the 3 checkboxes as they were (active, not active, active).

My specs:
Arch Linux, OpenJDK 8, i7 7700, 16gb RAM

Do you have any idea what's going on ?

Thanks for you time,

Regards

PS: I notice an error when launching Gisgraphy (either from launch.sh or start.jar):

```
WARN [main] JDBCExceptionReporter | SQL Error: 0, SQLState: 42703
ERROR [main] JDBCExceptionReporter | ERROR: column t1.tgconstrname does not exist
ERROR [main] SchemaUpdate | could not complete schema update
org.hibernate.exception.SQLGrammarException: could not get table metadata: ATM
```

...

```
Caused by: org.postgresql.util.PSQLException: ERROR: column t1.tgconstrname does not exist
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:1525)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1309)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:188)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:452)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:340)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:239)
        at org.postgresql.jdbc2.AbstractJdbc2DatabaseMetaData.getImportedExportedKeys(AbstractJdbc2DatabaseMetaData.java:3359)
        at org.postgresql.jdbc2.AbstractJdbc2DatabaseMetaData.getImportedKeys(AbstractJdbc2DatabaseMetaData.java:3552)
        at org.hibernate.tool.hbm2ddl.TableMetadata.initForeignKeys(TableMetadata.java:161)
        at org.hibernate.tool.hbm2ddl.TableMetadata.<init>(TableMetadata.java:60)
        at org.hibernate.tool.hbm2ddl.DatabaseMetadata.getTableMetadata(DatabaseMetadata.java:113)
```

but the service runs fine anyway afterwards and I can access every page, so I'm not sure it's related.

**Comment by gisgraphy** [comment]
I suspect that it is due to postgres 10.3. Gisgraphy has been tested on postgres 9.X.

see https://liquibase.jira.com/browse/CORE-3135

please try with postgres 9.X

**Comment by RaphaelFourdrilis** [comment]
Alright, I'll try that right now, thanks.

In the meantime, I see a guy solved the problem by updating the jdbc driver; could it be done easily from my end or does it have to be done on yours ?

**Comment by gisgraphy** [comment]
yes ! try to update the driver (it is also a solution). if it works, tell me which version have you used and I will do an update.

**Comment by RaphaelFourdrilis** [comment]
No luck with PostgresSql 9.6 and Postgis 2.3 :(

From the interface:

```
Error : An error occurred when processing GeonamesCountryImporter : An error occurred when processing countryInfo.txt : An Error occurred on Line 1 for AD	AND	020	AN	Andorra	Andorra la Vella	468	84000	EU	.ad	EUR	Euro	376	AD###	^(?:AD)*(\d{3})$	ca	3041565	ES,FR	: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.Gis
```

From the logs folder:

```
INFO [importBackgroundThread] ImporterManager | will now process GeonamesCountryImporter
ERROR [importBackgroundThread] BasicPropertyAccessor | IllegalArgumentException in class: com.gisgraphy.domain.geoloc.entity.GisFeature, setter method of property: adm
ERROR [importBackgroundThread] BasicPropertyAccessor | expected type: com.gisgraphy.domain.geoloc.entity.Adm, actual value: java.lang.Boolean
ERROR [importBackgroundThread] AbstractSimpleImporterProcessor | An error occurred when processing GeonamesCountryImporter : An error occurred when processing countryInfo.txt : An Error occurred on Line 1 for AD     AND     020     AN  Andorra  Andorra la Vella        468     84000   EU      .ad     EUR     Euro    376     AD###   ^(?:AD)*(\d{3})$        ca      3041565 ES,FR    : IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
com.gisgraphy.importer.ImporterException: An error occurred when processing countryInfo.txt : An Error occurred on Line 1 for AD        AND     020     AN      Andorra Andorra la Vella        468     84000   EU      .ad     EUR     Euro376      AD###   ^(?:AD)*(\d{3})$        ca      3041565 ES,FR    : IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.processFile(AbstractSimpleImporterProcessor.java:422)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.process(AbstractSimpleImporterProcessor.java:306)
        at com.gisgraphy.importer.ImporterManager.importAll(ImporterManager.java:138)
        at com.gisgraphy.webapp.action.ImportAction.doImport(ImportAction.java:130)
        at com.gisgraphy.webapp.action.ImportAction.execute(ImportAction.java:165)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.opensymphony.xwork2.DefaultActionInvocation.invokeAction(DefaultActionInvocation.java:441)
        at com.opensymphony.xwork2.DefaultActionInvocation.invokeActionOnly(DefaultActionInvocation.java:280)
        at org.apache.struts2.interceptor.BackgroundProcess$1.run(BackgroundProcess.java:57)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.orm.hibernate3.HibernateSystemException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at org.springframework.orm.hibernate3.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:709)
        at org.springframework.orm.hibernate3.HibernateAccessor.convertHibernateAccessException(HibernateAccessor.java:416)
        at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:416)
        at org.springframework.orm.hibernate3.HibernateTemplate.executeWithNativeSession(HibernateTemplate.java:379)
        at org.springframework.orm.hibernate3.HibernateTemplate.saveOrUpdate(HibernateTemplate.java:782)
        at com.gisgraphy.domain.repository.GenericDao.save(GenericDao.java:112)
        at com.gisgraphy.domain.repository.GenericGisDao.save(GenericGisDao.java:424)
        at com.gisgraphy.domain.repository.CountryDao.save(CountryDao.java:72)
        at com.gisgraphy.domain.repository.CountryDao.save(CountryDao.java:46)
        at com.gisgraphy.importer.GeonamesCountryImporter.processData(GeonamesCountryImporter.java:162)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.readLineAndProcessData(AbstractSimpleImporterProcessor.java:235)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.processFile(AbstractSimpleImporterProcessor.java:402)
        ... 12 more
Caused by: org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at org.hibernate.property.BasicPropertyAccessor$BasicSetter.set(BasicPropertyAccessor.java:128)
        at org.hibernate.tuple.entity.AbstractEntityTuplizer.setPropertyValues(AbstractEntityTuplizer.java:583)
        at org.hibernate.tuple.entity.PojoEntityTuplizer.setPropertyValues(PojoEntityTuplizer.java:229)
        at org.hibernate.persister.entity.AbstractEntityPersister.setPropertyValues(AbstractEntityPersister.java:3847)
        at org.hibernate.event.def.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:299)
        at org.hibernate.event.def.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:203)
        at org.hibernate.event.def.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:143)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.saveWithGeneratedOrRequestedId(DefaultSaveOrUpdateEventListener.java:210)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.entityIsTransient(DefaultSaveOrUpdateEventListener.java:195)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.performSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:117)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.onSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:93)
        at org.hibernate.impl.SessionImpl.fireSaveOrUpdate(SessionImpl.java:685)
        at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:677)
        at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:673)
        at org.springframework.orm.hibernate3.HibernateTemplate$16.doInHibernate(HibernateTemplate.java:786)
        at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:411)
        ... 21 more
Caused by: java.lang.IllegalArgumentException: argument type mismatch
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.hibernate.property.BasicPropertyAccessor$BasicSetter.set(BasicPropertyAccessor.java:66)
        ... 36 more
```

and then

```
[INFO] [com.gisgraphy.importer.ImporterManager] [26 Mar 2018 12:16:52,794] (importBackgroundThread,) : will now process GeonamesCountryImporter
[ERROR] [com.gisgraphy.importer.AbstractSimpleImporterProcessor] [26 Mar 2018 12:16:52,965] (importBackgroundThread,) : An error occurred when processing GeonamesCountryImporter : An error occurred when processing countryInfo.txt : An Error occurred on Line 1 for AD   AND     020     AN      Andorra Andorra la Vella        468     84000   EU      .ad     EUR     Euro    376     AD###   ^(?:AD)*(\d{3})$        ca      3041565 ES,FR    : IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
com.gisgraphy.importer.ImporterException: An error occurred when processing countryInfo.txt : An Error occurred on Line 1 for AD        AND     020     AN      Andorra Andorra la Vella        468     84000   EU      .ad     EUR     Euro376      AD###   ^(?:AD)*(\d{3})$        ca      3041565 ES,FR    : IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.processFile(AbstractSimpleImporterProcessor.java:422)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.process(AbstractSimpleImporterProcessor.java:306)
        at com.gisgraphy.importer.ImporterManager.importAll(ImporterManager.java:138)
        at com.gisgraphy.webapp.action.ImportAction.doImport(ImportAction.java:130)
        at com.gisgraphy.webapp.action.ImportAction.execute(ImportAction.java:165)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.opensymphony.xwork2.DefaultActionInvocation.invokeAction(DefaultActionInvocation.java:441)
        at com.opensymphony.xwork2.DefaultActionInvocation.invokeActionOnly(DefaultActionInvocation.java:280)
        at org.apache.struts2.interceptor.BackgroundProcess$1.run(BackgroundProcess.java:57)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.orm.hibernate3.HibernateSystemException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm; nested exception is org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at org.springframework.orm.hibernate3.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:709)
        at org.springframework.orm.hibernate3.HibernateAccessor.convertHibernateAccessException(HibernateAccessor.java:416)
        at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:416)
        at org.springframework.orm.hibernate3.HibernateTemplate.executeWithNativeSession(HibernateTemplate.java:379)
        at org.springframework.orm.hibernate3.HibernateTemplate.saveOrUpdate(HibernateTemplate.java:782)
        at com.gisgraphy.domain.repository.GenericDao.save(GenericDao.java:112)
        at com.gisgraphy.domain.repository.GenericGisDao.save(GenericGisDao.java:424)
        at com.gisgraphy.domain.repository.CountryDao.save(CountryDao.java:72)
        at com.gisgraphy.domain.repository.CountryDao.save(CountryDao.java:46)
        at com.gisgraphy.importer.GeonamesCountryImporter.processData(GeonamesCountryImporter.java:162)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.readLineAndProcessData(AbstractSimpleImporterProcessor.java:235)
        at com.gisgraphy.importer.AbstractSimpleImporterProcessor.processFile(AbstractSimpleImporterProcessor.java:402)
        ... 12 more
Caused by: org.hibernate.PropertyAccessException: IllegalArgumentException occurred while calling setter of com.gisgraphy.domain.geoloc.entity.GisFeature.adm
        at org.hibernate.property.BasicPropertyAccessor$BasicSetter.set(BasicPropertyAccessor.java:128)
        at org.hibernate.tuple.entity.AbstractEntityTuplizer.setPropertyValues(AbstractEntityTuplizer.java:583)
        at org.hibernate.tuple.entity.PojoEntityTuplizer.setPropertyValues(PojoEntityTuplizer.java:229)
        at org.hibernate.persister.entity.AbstractEntityPersister.setPropertyValues(AbstractEntityPersister.java:3847)
        at org.hibernate.event.def.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:299)
        at org.hibernate.event.def.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:203)
        at org.hibernate.event.def.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:143)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.saveWithGeneratedOrRequestedId(DefaultSaveOrUpdateEventListener.java:210)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.entityIsTransient(DefaultSaveOrUpdateEventListener.java:195)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.performSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:117)
        at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.onSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:93)
        at org.hibernate.impl.SessionImpl.fireSaveOrUpdate(SessionImpl.java:685)
        at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:677)
        at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:673)
        at org.springframework.orm.hibernate3.HibernateTemplate$16.doInHibernate(HibernateTemplate.java:786)
        at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:411)
        ... 21 more
Caused by: java.lang.IllegalArgumentException: argument type mismatch
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.hibernate.property.BasicPropertyAccessor$BasicSetter.set(BasicPropertyAccessor.java:66)
        ... 36 more
```

TLDR;

`ERROR [importBackgroundThread] BasicPropertyAccessor | expected type: com.gisgraphy.domain.geoloc.entity.Adm, actual value: java.lang.Boolean`

**Comment by gisgraphy** [comment]
nothing to see with the first error. It is a well-known bug with the JVM,( not PostgreSQL anymore) that I have never reproduced on my servers. maybe a try with sun JVM could resolve the problem.

if it works (java problem is solved), please try with PostgreSQL 10.X driver to see if it solve the initial problem.

**Comment by RaphaelFourdrilis** [comment]
Yeah sorry I forgot to mention:

The original bug is gone, with postgres 9.6 and postgis 2.3. This is indeed a different bug.

I'm trying to change the driver right now. As I understand it, i'd just have to replace:

`webapps/ROOT/WEB-INF/lib/spring-jdbc-4.3.5.RELEASE.jar` by a more recent one.

Is this the only thing I should change, or should I replace

`webapps/ROOT/WEB-INF/lib/postgis-jdbc-extensions-1.1.2-1.jar` and `webapps/ROOT/WEB-INF/lib/postgresql-8.1-407.jdbc3.jar` too ?

**Comment by gisgraphy** [comment]
I would say all the files mentioned.

**Comment by RaphaelFourdrilis** [comment]
Trying that right now with latest versions of each jar from maven.

**Comment by RaphaelFourdrilis** [comment]
Whoopteedoo!

Postgresql 10.3, Postgis 2.4, Oracle JDK 8 seems to work! The import went further than ever before, so I'm going to say it is resolved.

What I changed:

I replaced these:

```
spring-beans-4.3.5.RELEASE.jar
spring-core-4.3.5.RELEASE.jar
spring-jcl-4.3.5.RELEASE.jar
spring-jdbc-4.3.5.RELEASE.jar
spring-tx-4.3.5.RELEASE.jar

postgresql-8.1-407.jdbc3.jar
```

by their updated versions:

```
spring-beans-5.0.4.RELEASE.jar
spring-core-5.0.4.RELEASE.jar
spring-jcl-5.0.4.RELEASE.jar
spring-jdbc-5.0.4.RELEASE.jar
spring-tx-5.0.4.RELEASE.jar

postgresql-42.2.2.jar
```

which I got from this (quite useful) website:

`https://jar-download.com/`

and everything seems to work fine.

Thanks again for your help, and sorry for the inconvenience.

**Comment by ecogent** [comment]
better files here:

sudo wget http://central.maven.org/maven2/org/springframework/spring-beans/5.0.4.RELEASE/spring-beans-5.0.4.RELEASE.jar

sudo wget http://central.maven.org/maven2/org/springframework/spring-core/5.0.4.RELEASE/spring-core-5.0.4.RELEASE.jar

sudo wget http://central.maven.org/maven2/org/springframework/spring-jcl/5.0.4.RELEASE/spring-jcl-5.0.4.RELEASE.jar

sudo wget http://central.maven.org/maven2/org/springframework/spring-jdbc/5.0.4.RELEASE/spring-jdbc-5.0.4.RELEASE.jar

sudo wget http://central.maven.org/maven2/org/springframework/spring-tx/5.0.4.RELEASE/spring-tx-5.0.4.RELEASE.jar

sudo wget http://central.maven.org/maven2/org/postgresql/postgresql/42.2.2/postgresql-42.2.2.jar

**Comment by jcustovic** [comment]
Hi,

I am having the exact same issue and cannot solve it with the above suggestions.
I am running Ubuntu 18.04, Postgresql 9.6, Postigs 2.3 and Oracle Java JDK 1.8.0_191.

Any other ideas on how to solve it?

>  An error occurred when processing GeonamesCountryImporter : An error occurred when processing countryInfo.txt...
> Caused by: java.lang.IllegalArgumentException: argument type mismatch

Br,
J
